### PR TITLE
Use PyTorch's SiLU in `nnet.activations.Swish`

### PR DESCRIPTION
**Pull request by asumagic**
## What does this PR do?

The result is a more fused kernel with a simple `beta == 1` fast path. The result is 2x (beta == 1) ~ 3x (beta != 1) the performance, tested on ROCm on a 7900XT, 256MiB random test tensor (might be a bit different on CUDA).

Swish is usually not very significant in execution time but this is simple enough anyway.

Our Swish is `x * sigmoid(beta * x)`, torch's SiLU is `x * sigmoid(x)`. Simply reimplement it as `SiLU(x * beta)` and skip the beta multiplication if it's `1.0` (commonly used, and the default).

Tested using:

```python
import torch

class OldSwish(torch.nn.Module):
    def __init__(self, beta=1):
        super().__init__()
        self.beta = beta
        self.sigmoid = torch.nn.Sigmoid()

    def forward(self, x):
        return x * self.sigmoid(self.beta * x)

old_swish = OldSwish()

class NewSwish(torch.nn.Module):
    def __init__(self, beta: float = 1.0):
        super().__init__()
        self.beta = beta
        self.silu = torch.nn.SiLU()

    def forward(self, x):
        if self.beta != 1:  # fast path
            x = x * self.beta

        return self.silu(x)

new_swish = NewSwish(1.0)

test_tensor = torch.rand((256, 256, 1024), device="cuda")

import time

def bench(swish_fn, x):
    torch.cuda.synchronize()

    start = time.time()
    for i in range(1000):
        y = swish_fn(x)
    torch.cuda.synchronize()
    end = time.time()

    print(end-start)

    return y

yold = bench(old_swish, test_tensor)  # 2.798469305038452
ynew = bench(new_swish, test_tensor)  # 0.7957239151000977
print(torch.allclose(yold, ynew))     # True

new_swish_custom_beta = NewSwish(1.5)
bench(new_swish_custom_beta, test_tensor)  # 1.590627908706665
```

<details>
  <summary><b>Before submitting</b></summary>

- [ ] Did you read the [contributor guideline](https://speechbrain.readthedocs.io/en/latest/contributing.html)?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing [tests](https://github.com/speechbrain/speechbrain/tree/develop/tests) pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Does your code adhere to project-specific code style and conventions?

</details>

## PR review

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
- [ ] Confirm that the changes adhere to compatibility requirements (e.g., Python version, platform)
- [ ] Review the self-review checklist to ensure the code is ready for review

</details>

### CODE REVIEW

**Comment by asumagic** [comment]
CI is failing due to `https://github.com/huggingface/huggingface_hub/issues/2186` (not linking to avoid spamming them which they're getting plenty of), will wait for the issue to be resolved upstream.

**Comment by TParcollet** [comment]
This looks good, I personally removed the switch from some of my models due to this performance issue. Is this true for CUDA enable GPU as well?

**Comment by asumagic** [comment]
> This looks good, I personally removed the switch from some of my models due to this performance issue. Is this true for CUDA enable GPU as well?

On a 2080Ti x1000 with 256MiB tensors fp32:

- old 3.7351620197296143s
- new (beta == 1): 1.0813236236572266s
- new (beta != 1): 2.0165617465972900s

On a 2080Ti x1000 with 128MiB tensors fp16:

- old 1.7387142181396484
- new (beta == 1): 0.5031118392944336
- new (beta != 1): 1.0063819885253906

On a 2080Ti x1000 with 4MiB tensors fp16:

- old 0.060301780700683594s
- new (beta == 1): 0.01763296127319336s
- new (beta != 1): 0.03473210334777832s

So there seems to be a very consistent improvement in any case (which makes sense since this is probably pretty memory bandwidth bound).

**Review by Adel-Moumen — APPROVED** [review]
LGTM.
